src/blant-sampling.c

```diff
@@ -22,82 +22,96 @@ int *MCMCGetNeighbor(int *Xcurrent, GRAPH *G)
 {
     if (mcmc_d == 2)
     {
-	int oldu = Xcurrent[0];
-	int oldv = Xcurrent[1];
-	while (oldu == Xcurrent[0] && oldv == Xcurrent[1]) {
-	    double p = RandomUniform();
-	    // if 0 < p < 1, p < deg(u) + deg(v) then
-	    if (p < ((double)GraphDegree(G,Xcurrent[0])/(GraphDegree(G,Xcurrent[0]) + GraphDegree(G,Xcurrent[1])))) {
-		// select randomly from Neigh(u) and swap
-		Xcurrent[1] = GraphRandomNeighbor(G, Xcurrent[0]);
-	    }
-	    else {
-		// select randomly from Neigh(v) and swap
-		Xcurrent[0] = GraphRandomNeighbor(G,Xcurrent[1]);
-	    }
-	    if(_sampleSubmethod == SAMPLE_MCMC_EC) {
-		if(!GraphAreConnected(_EDGE_COVER_G, Xcurrent[0], Xcurrent[1]) && RandomUniform() < 0.9) { // undo the attempt
-		    Xcurrent[0] = oldu;
-		    Xcurrent[1] = oldv;
-		}
-	    }
-	}
+        int oldu = Xcurrent[0];
+        int oldv = Xcurrent[1];
+        while (oldu == Xcurrent[0] && oldv == Xcurrent[1])
+        {
+            double p = RandomUniform();
+            // if 0 < p < 1, p < deg(u) + deg(v) then
+            if (p < ((double)GraphDegree(G, Xcurrent[0]) / (GraphDegree(G, Xcurrent[0]) + GraphDegree(G, Xcurrent[1]))))
+            {
```

Comment:
Also please put my brackets back on the line above... I know yours looks better but I prefer to minimize the vertical space used... and also because it makes it more difficult for me to see what's *actually* changed. :-/

Reply:
Fixed with latest force push